### PR TITLE
Bug 1468159 - convert _.union() usages to ES6

### PR DESCRIPTION
**Pull request by eliottvincent**

### CODE REVIEW

**Comment by eliottvincent** [comment]
Travis failed while creating a db. @camd can you please trigger it again? Thanks 😊

**Comment by edmorley** [comment]
I retriggered previously to resolve the intermittent Travis failure, however since then this PR now has conflicts. Could you rebase this PR?

**Comment by eliottvincent** [comment]
Rebase done ✅

**Comment by eliottvincent** [comment]
Oh what's wrong? I rebased this branch onto origin/master using GitKraken.

**Comment by eliottvincent** [comment]
I was afraid of all the commits being replayed on the branch but it's the goal of a rebase right?

**Comment by eliottvincent** [comment]
I see. Is it the the same mistake for #3657 & #3632?

**Comment by edmorley** [comment]
Yeah it would be good to update the others too :-)
After making the changes locally, it will need a `git push --force` to update the PR.

**Comment by eliottvincent** [comment]
Ok does this look any good?

File: ui/js/controllers/perf/compare.js [hunk 1]
```diff
@@ -306,12 +306,12 @@ perf.controller('CompareResultsCtrl', [
...
                     }).then((newSeriesList) => {
-                        $scope.platformList = _.union(
-                            $scope.platformList,
-                            [...new Set(_.map(newSeriesList, 'platform'))]);
-                        $scope.testList = _.union(
-                            $scope.testList,
-                            [...new Set(_.map(newSeriesList, 'name'))]);
+                        $scope.platformList = [...new Set([
+                            ...$scope.platformList,
+                            ...[...new Set(_.map(newSeriesList, 'platform'))]])];
```
Comment (camd) [inline ui/js/controllers/perf/compare.js]: The double spread isn't needed here.  You can just do:

    ...new Set(_.map(newSeriesList, 'platform'))])];

Comment (camd) [reply ui/js/controllers/perf/compare.js]: So you can simplify the other 3 double-spreads as well.

File: ui/js/controllers/perf/compare.js [hunk 2]
```diff
@@ -772,12 +774,14 @@ perf.controller('CompareSubtestResultsCtrl', [
                                 parent_signature: $scope.newSignature,
                                 framework: $scope.filterOptions.framework,
                             }).then(function (newSeriesList) {
-                                $scope.platformList = [...new Set(_.union(
-                                    $scope.platformList,
-                                    newSeriesList.map(series => series.platform)))];
-                                $scope.testList = [...new Set(_.union(
-                                    $scope.testList,
-                                    newSeriesList.map(series => series.name)))];
+                                $scope.platformList = [...new Set([...new Set([
```
Comment (camd) [inline ui/js/controllers/perf/compare.js]: Sorry I didn't see this before.  We also don't need double ``new Set``.  A single version of that will do.

File: ui/js/controllers/perf/compare.js [hunk 3]
```diff
@@ -772,12 +774,14 @@ perf.controller('CompareSubtestResultsCtrl', [
...
+                                    ...$scope.platformList,
+                                    ...newSeriesList.map(series => series.platform),
+                                ])])];
+                                $scope.testList = [...new Set([...new Set([
```
Comment (camd) [inline ui/js/controllers/perf/compare.js]: remove double ``new Set``

File: ui/js/controllers/perf/compare.js [hunk 4]
```diff
@@ -839,12 +843,14 @@ perf.controller('CompareSubtestResultsCtrl', [
                                     parent_signature: $scope.newSignature,
                                     framework: $scope.filterOptions.framework,
                                 }).then(function (newSeriesList) {
-                                    $scope.platformList = [...new Set(_.union(
-                                        $scope.platformList,
-                                        newSeriesList.map(series => series.platform)))];
-                                    $scope.testList = [...new Set(_.union(
-                                        $scope.testList,
-                                        newSeriesList.map(series => series.name)))];
+                                    $scope.platformList = [...new Set([...new Set([
```
Comment (camd) [inline ui/js/controllers/perf/compare.js]: remove double ``new Set``

File: ui/js/controllers/perf/alerts.js [hunk 5]
```diff
@@ -491,8 +491,9 @@ perf.controller('AlertsCtrl', [
                 if (_.isUndefined($scope.alertSummaries)) {
                     $scope.alertSummaries = alertSummaries;
                 } else {
-                    $scope.alertSummaries = _.union($scope.alertSummaries,
```
Comment (camd) [inline ui/js/controllers/perf/alerts.js]: This is throwing an error when I go to the perfherder Alerts tab when pointed to stage (``yarn start:stage``).  Can you test this out and see why it's erroring?  Perhaps one of those objects are not in the form we expect, or null?

Comment (eliottvincent) [reply ui/js/controllers/perf/alerts.js]: Sorry, I must have missed that one. I'll take a look.

Comment (eliottvincent) [reply ui/js/controllers/perf/alerts.js]: I assume it's the same case as #3657? I don't see any error in the Alert tab.

**Comment by eliottvincent** [comment]
Ok let's finish this one. I rebased once again and haven't encountered any error.
@camd can you review on your side and confirm? Thanks!

**Pull request merged**